File: Remove_nth_from_end.py
#Time Complexity:O(n)
#Space Complexity:O(1)
#leetcode submission:succesful
#use of fast and slow pointer 
#the fast pointer is n postion forward

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        temp2=head
        temp1=head
        while(n>0):
            temp2=temp2.next
            n=n-1
        if not temp2:
            return head.next
        while(temp2.next!=None):
            temp2=temp2.next
            temp1=temp1.next
        temp1.next=temp1.next.next
        return head

File: RemoveNthFromEnd.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LinkedList
{
    class RemoveNthFromEndLC
    {
        // Definition for singly-linked list.
        public class ListNode
        {
            public int val;
            public ListNode next;
            public ListNode(int val = 0, ListNode next = null)
            {
                this.val = val;
                this.next = next;
            }
        }

        public ListNode RemoveNthFromEnd(ListNode head, int n)
        {
            if (head == null)
            {
                return head;
            }
            ListNode dummy = new ListNode(-1);
            dummy.next = head;
            ListNode slow = dummy, fast = dummy;
            int cnt = 0;
            while (cnt <= n)
            {
                fast = fast.next;
                cnt++;
            }
            while (fast != null)
            {
                fast = fast.next;
                slow = slow.next;
            }
            slow.next = slow.next.next;
            return dummy.next;
        }
    }
}
